### PR TITLE
Modify job to publish clang-tidy binaries on S3

### PR DESCRIPTION
**Pull request by 1ntEgr8**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48 Modify job to publish clang-tidy binaries on S3**
* #47 Statically link clang-tidy binary

This PR modifies the clang-tidy docker workflow to upload the freshly built binary to an S3 bucket.

Test Plan:
#49

### CODE REVIEW

File: .github/workflows/push_ci_docker_base.yml [hunk 1]
```diff
@@ -21,4 +21,22 @@ jobs:
...
+          docker rm -v $image_id
+      - uses: driazati/upload-artifact-s3@7faed57ea7f289192a68d1af3eb6358785e21b86
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        name: Publish binary
+        with:
+          name: clang-tidy
+          if-no-files-found: error
+          s3-prefix: linux64
+          s3-bucket: oss-clang-format
+          path: ./clang-tidy
```
Comment (driazati) [inline .github/workflows/push_ci_docker_base.yml:40]: I'm not sure how [s3 interprets `./` in prefixes](https://alexwlchan.net/2020/08/s3-keys-are-not-file-paths/) (the upload-artifact-s3 action uses [`path` directly](https://github.com/seemethere/upload-artifact-s3/compare/main...driazati:main#diff-519fa009c5415a88d003e807b2f95d82d6acdd9651c8352aa7f6f9f808c44a0fL64) to create the complete s3 name), but it's all the same so we should just make this be

```suggestion
          path: clang-tidy
```

File: .github/workflows/push_ci_docker_base.yml [hunk 2]
```diff
@@ -21,4 +21,22 @@ jobs:
...
-
+      - name: Extract binary
+        run: |
+          set -ex
+          # Copying files directly from a docker image is not supported
+          # As a workaround, we create a temporary container, copy the binary, and remove it
+          image_id=$(docker create ghcr.io/pytorch/cilint-clang-tidy:$GITHUB_SHA)
+          docker cp $image_id:/clang-tidy-checks/build/bin/clang-tidy ./clang-tidy
+          docker rm -v $image_id
+      - uses: driazati/upload-artifact-s3@7faed57ea7f289192a68d1af3eb6358785e21b86
```
[the remark this answers is not in the document]
Comment (1ntEgr8) [reply .github/workflows/push_ci_docker_base.yml:32]: `seemethere/upload-artifact-s3` doesn't have the `aws-access-key-id`, `aws-secret-access-key`, and `s3-prefix` inputs

**Review by driazati — APPROVED** [review]

File: .github/workflows/push_ci_docker_base.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (driazati) [reply .github/workflows/push_ci_docker_base.yml]: We should definitely merge the two https://github.com/pytorch/pytorch/issues/60999

**Comment by driazati** [comment]
whoops, should have `ghstack land`ed this instead of merging.

fixed in 8bbd045f73e5787efa78d65112988e1f07be3b25

**Pull request merged**